utils/yoke/src/zero_copy_from.rs

```diff
@@ -192,3 +192,39 @@ impl ZeroCopyFrom<String> for &'static str {
         cart
     }
 }
+
+impl<C, T: ZeroCopyFrom<C>> ZeroCopyFrom<Option<C>> for Option<T> {
+    fn zero_copy_from<'b>(cart: &'b Option<C>) -> Option<<T as Yokeable<'b>>::Output> {
+        cart.as_ref()
+            .map(|c| <T as ZeroCopyFrom<C>>::zero_copy_from(c))
+    }
+}
+
+impl<C1, T1: ZeroCopyFrom<C1>, C2, T2: ZeroCopyFrom<C2>> ZeroCopyFrom<(C1, C2)> for (T1, T2) {
+    fn zero_copy_from<'b>(
+        cart: &'b (C1, C2),
+    ) -> (<T1 as Yokeable<'b>>::Output, <T2 as Yokeable<'b>>::Output) {
+        (
+            <T1 as ZeroCopyFrom<C1>>::zero_copy_from(&cart.0),
+            <T2 as ZeroCopyFrom<C2>>::zero_copy_from(&cart.1),
+        )
+    }
+}
+
+// These duplicate the functionality from above and aren't quite necessary due
+// to deref coercions, however for the custom derive to work, there always needs
+// to be `impl ZCF<T> for T`, otherwise it may fail to perform the necessary
+// type inference. Deref coercions do not typically work when sufficient generics
+// or inference are involved, and the proc macro does not necessarily have
+// enough type information to figure this out on its own.
+impl<B: ToOwned + ?Sized + 'static> ZeroCopyFrom<Cow<'_, B>> for Cow<'static, B> {
```

Comment:
Question: I'm surprised the compiler lets these get through instead of throwing a multiple definition error.  Why?

Reply:
it's a different trait each time, `ZeroCopyFrom<something different>`

Reply:
I see, for `Cow<'static, str>`, this definition comes from `Cow<'_, str>`, whereas the ones above come from `&str`.